crux-core/src/crux/node.clj

```diff
@@ -123,26 +123,31 @@
            (kv/get-value (kv/new-snapshot kv-store)
                          (c/encode-failed-tx-id-key-to nil tx-id)))))))
 
-  (newTxLogContext [this]
+  (openTxLogIterator ^TxLogIterator [this from-tx-id with-ops?]
     (cio/with-read-lock lock
       (ensure-node-open this)
-      (db/new-tx-log-context tx-log)))
-
-  (txLog [this tx-log-context from-tx-id with-ops?]
-    (cio/with-read-lock lock
-      (ensure-node-open this)
-      (for [{:keys [crux.tx/tx-id
-                    crux.tx.event/tx-events] :as tx-log-entry} (db/tx-log tx-log tx-log-context from-tx-id)
-            :when (with-open [snapshot (kv/new-snapshot kv-store)]
-                    (nil? (kv/get-value snapshot (c/encode-failed-tx-id-key-to nil tx-id))))]
-        (if with-ops?
-          (-> tx-log-entry
-              (dissoc :crux.tx.event/tx-events)
-              (assoc :crux.api/tx-ops
-                     (with-open [snapshot (kv/new-snapshot kv-store)]
-                       (->> tx-events
-                            (mapv #(tx/tx-event->tx-op % snapshot object-store))))))
-          tx-log-entry))))
+      (let [tx-log-iterator (db/open-tx-log-iterator tx-log from-tx-id)
+            snapshot (kv/new-snapshot kv-store)
+            filtered-txs (filter
```

Comment:
we can make this clearer to the reader with some threading - we're essentially piping one sequence through a few operations but the current code layout jumps around a little in terms of when the various operations happen:

```clojure
(let [tx-log (-> (iterator-seq tx-log-iterator)
                 (->> (filter #(nil? (...))))
                 (cond->> with-ops? (map (fn [...]
                                           ...))))]
  (db/->closeable-tx-log-iterator ...))
```